tools/haddock-score.py

```diff
@@ -0,0 +1,197 @@
+    topo_out_file_name.unlink()
+    topo_err_file_name.unlink()
+    tidy_model_path.unlink()
+
+    return pdb_obj
+
+
+def emscoring_wrapper(pdb_object, root="calc-hs", model_num=1):
+    """
+    Wrapper for the EMScoring.
+
+    Parameters
+    ----------
+    pdb_object : PDBFile
+        The PDBFile object.
+
+    root : str
+        The root of the output PDB name.
+
+    model_num : int
+        The model number.
+
+    Returns
+    -------
+    component_dic : dict
+        The dictionary with the energetic components.
+    """
+
+    main_emscoring_cns_script_as_string = Path(
+        emscoring_module_folder, "cns/emscoring.cns"
+        ).read_text()
+
+    scoring_inp_file_name = prepare_cns_input(
+        1,
+        pdb_object,
+        Path.cwd(),
+        main_emscoring_cns_script_as_string,
+        read_from_yaml_config(SCORING_DEFAULT_CONFIG),
+        root,
+        native_segid=True,
+        )
+    env = {
+        "MODULE": Path(emscoring_module_folder, "cns").resolve(),
+        "MODDIR": ".",
+        "TOPPAR": TOPPAR_DIR,
+        }
+    scoring_out_file_name = scoring_inp_file_name.with_suffix(".out")
+    scoring_err_file_name = scoring_inp_file_name.with_suffix(".err")
+
+    inp = open(scoring_inp_file_name, "r")
+    out = open(scoring_out_file_name, "w")
+    err = open(scoring_err_file_name, "w")
+    _ = subprocess.run(CNS_EXEC, env=env, stdin=inp, stdout=out, stderr=err)
+
+    output_pdb = Path(Path.cwd(), f"{root}_{model_num}.pdb").resolve()
+
+    component_dic = HaddockModel(output_pdb).energies
+
+    # Clean everything except the minimized pdb
+    scoring_out_file_name.unlink()
+    scoring_err_file_name.unlink()
+    scoring_inp_file_name.unlink()
+
+    pdb_object.rel_path.unlink()
+    pdb_object.topology.rel_path.unlink()
+
+    return component_dic
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        prog="haddock3-score",
+        description=__doc__,
+        )
+
+    parser.add_argument("pdb_file")
+
+    args = parser.parse_args()
+
+    # Create the topology and return a PDBFile object
+    pdb_object = topo_wrapper(args.pdb_file)
+
+    # Run emscoring and return the energy components
+    haddock_score_component_dic = emscoring_wrapper(pdb_object)
+
+    # Get the components
+    vdw = haddock_score_component_dic["vdw"]
+    elec = haddock_score_component_dic["elec"]
+    desolv = haddock_score_component_dic["desolv"]
+    air = haddock_score_component_dic["air"]
+    # bsa = haddock_score_component_dic["bsa"]
+
+    # emscoring is equivalent to itw
+    haddock_score_itw = (1.0 * vdw) + (0.2 * elec) + \
+        (1.0 * desolv) + (0.1 * air)
+
+    print("-----")
+    print(f"vdw\t{vdw:.4f}")
+    print(f"elec't{elec:.4f}")
+    print(f"desolv\t{desolv:.4f}")
+    print(f"air\t{air:.4f}")
+    print("HADDOCK-score = (1.0 * vdw) + (0.2 * elec) + (1.0 * desolv) + (0.1 * air)")
+    print("-----")
+    print(f"HADDOCK-score (emscoring) = {haddock_score_itw:.4f}")
```

Comment:
air will be in principle 0
may-be more interesting to output the BSA.
And I would not output everything by default, but only as an option (e.g. if the `-full` argument is given). 
Simpler for incorporating the score in other scripts.

Reply:
And same thing for outputting the `pdb` and `psf` files, only if an option is given, e.q. `-outpdb` -`outpsf`

Reply:
> air will be in principle 0
> may-be more interesting to output the BSA.
> And I would not output everything by default, but only as an option (e.g. if the `-full` argument is given).
> Simpler for incorporating the score in other scripts.


Added `--full` option


> And same thing for outputting the `pdb` and `psf` files, only if an option is given, e.q. `-outpdb` -`outpsf`


Added `--outputpdb` and `--outputpsf` options